Formula/crystal.rb

```diff
@@ -117,9 +123,19 @@ def install
 
     # Build shards (with recently built crystal)
     resource("shards").stage do
-      system "make", "bin/shards", "CRYSTAL=#{buildpath}/bin/crystal",
-                                   "SHARDS=false",
-                                   *release_flags
+      shard_lock = YAML.load_file("shard.lock")
+      required_molinillo_version = shard_lock.dig("shards", "molinillo", "version")
+      available_molinillo_version = resource("molinillo").version.to_s
+      raise "`molinillo` resource is outdated!" unless required_molinillo_version == available_molinillo_version
```

Comment:
Should we use `odie` here?
```suggestion
      odie "`molinillo` resource is outdated!" unless required_molinillo_version == available_molinillo_version
```

At least we mention it in the formula cookbook https://docs.brew.sh/Formula-Cookbook#messaging